osquery/core/watcher.cpp

```diff
@@ -306,12 +306,30 @@ void WatcherRunner::watchExtensions() {
   for (const auto& extension : watcher_->extensions()) {
     // Check the extension status, causing a wait.
     int process_status = 0;
-    extension.second->checkStatus(process_status);
+    ProcessState status = extension.second->checkStatus(process_status);
 
-    auto ext_valid = extension.second->isValid();
+    bool ext_valid = (PROCESS_STILL_ALIVE == status);
     auto s = isChildSane(*extension.second);
 
-    if (!ext_valid || (!s.ok() && getUnixTime() >= delayedTime())) {
+    /*
```

Comment:
Do you mind moving this comment block into the docs or formatting it differently? I ask because we don't tend to write out cases like this in comments.